src/main/java/school/hei/patrimoine/visualisation/swing/ihm/MultiScreenApp.java

```diff
@@ -0,0 +1,216 @@
+    JButton removeButton = new JButton("Remove Input");
+    removeButton.addActionListener(
+        e -> {
+          if (inputFields.size() > MIN_INPUTS) {
+            removeInputField();
+          } else {
+            showMessageDialog(inputFrame, "At least 1 input is required.");
+          }
+        });
+    return removeButton;
+  }
+
+  private JButton newAddButton() {
+    JButton addButton = new JButton("Add Input");
+    addButton.addActionListener(
+        e -> {
+          if (inputFields.size() < MAX_INPUTS) {
+            addInputField();
+          } else {
+            showMessageDialog(inputFrame, "Maximum of 9 inputs allowed.");
+          }
+        });
+    return addButton;
+  }
+
+  private void addInitialInput() {
+    JTextField initialField = newGoogleDocsLinkTextField();
+    inputFields.add(initialField);
+    inputPanel.add(initialField);
+  }
+
+  private JTextField newGoogleDocsLinkTextField() {
+    var textField = new JTextField(20);
+    textField.setInputVerifier(linkIdInputVerifier);
+    return textField;
+  }
+
+  private void addInputField() {
+    JTextField newField = newGoogleDocsLinkTextField();
+    inputFields.add(newField);
+    inputPanel.add(newField);
+    inputFrame.pack();
+  }
+
+  private void removeInputField() {
+    JTextField fieldToRemove = inputFields.getLast();
+    inputPanel.remove(fieldToRemove);
+    inputFields.remove(fieldToRemove);
+    inputFrame.pack();
+  }
+
+  private void loadDataInBackground() {
+    JDialog loadingDialog = new JDialog(inputFrame, "Processing", true);
+    JLabel loadingLabel = new JLabel("Processing, please wait...");
+    loadingLabel.setHorizontalAlignment(SwingConstants.CENTER);
+    loadingDialog.getContentPane().add(loadingLabel, CENTER);
+    loadingDialog.setSize(300, 100);
+    loadingDialog.setLocationRelativeTo(inputFrame);
+
+    SwingWorker<List<Patrimoine>, Void> worker =
+        new SwingWorker<>() {
+          @Override
+          protected List<Patrimoine> doInBackground() throws Exception {
+            var ids = extractInputIds();
+            List<String> codePatrimoinesVisualisables = new ArrayList<>();
+            for (var id : ids) {
+              var code = googleApi.readDocsContent(authDetails, id);
+              codePatrimoinesVisualisables.add(code);
+            }
+            List<Patrimoine> patrimoinesVisualisables = new ArrayList<>();
+            Pattern pattern = Pattern.compile("public class (\\w+)");
+            for (String codePatrimoine : codePatrimoinesVisualisables) {
+              Matcher matcher = pattern.matcher(codePatrimoine);
+
+              if (matcher.find()) {
+                String className = matcher.group(1);
+
+                Patrimoine patrimoineVisualisable =
+                    PatrimoineCompiler.stringCompiler(className, codePatrimoine);
+                patrimoinesVisualisables.add(patrimoineVisualisable);
+              }
+            }
+            return patrimoinesVisualisables;
```

Comment:
to achieve better performance, Pattern should be compiled as a class constant and re-used as it is both thread-safe and immutable

Reply:
also, the logic to find the className clearly does not belong to this class hence, you should create another class  to do it or change the readContent model from the googleApi so that it includes both the code and the className